app/controllers/announcements_controller.rb

```diff
@@ -0,0 +1,54 @@
+class AnnouncementsController < ApplicationController
+  before_action :set_announcement, only: %i[edit update show destroy]
+
+  load_and_authorize_resource
+
+  def index
+    @episode = Episode.find(params['episode_id'])
+  end
+
+  def show; end
+
+  def new
+    @announcement = Announcement.new
+  end
+
+  def create
+    announcement = Announcement.new(announcement_params)
```

Comment:
`announcement.episode_id = params[:episode_id]`

Reply:
??????????????
I think announcement belongs to episode, so
```ruby
def create
  @episode.announcements.create(params)
end
...
def set_episode
  @episode = Episode.find(params['episode_id'])
end
...

```! never made things like that `announcement.episode_id = params[:episode_id]`!!!